### ISSUE TITLE
Modal functionality not configurable dynamically. No way to use element for multiple dialogs.

### ISSUE THREAD

**Issue by liechtym** [jshcrowthe/dialog-el#22]
Because the 'created' callback sets the modal/dialog's template, it is impossible to set the modal property dynamically on dialog-el.

For example, this would not work:

```html
<button id="button">Open Modal</button>
<dialog-el id="modal"></dialog-el>

<script>
  var modal = document.querySelector('#modal');
  var someCondition = *smart algorithm*;
  // Doesn't work because dialog-el's 'createdCallback' has already initialized the template
  if (someCondition) modal.setAttribute('modal', '');
  document.querySelector('#button').addEventListener('click', function(event){
    modal.show();
  });
</script>
```

This would not work either:
```html
<div id="container">
</div>

<script>
  var modal = document.createElement('dialog-el');
  //Does not work because element's "createdCallback" already fired.
  modal.setAttribute('modal', '');
  var cont = document.querySelector('#container');
  cont.appendChild(modal);
</script>
```
This leaves only one option:
```html
<dialog-el id="modal" modal></dialog-el>
```

Furthermore, I have often found it most efficient to use only one dialog element to handle the whole page's dialogs and/or modals. Because there is not a way to update the content and template of dialog-el after the created callback, this is impossible. Perhaps a method could be added, such as `el.updateTemplate('[modal/non-modal]')` or `el.updateContent(content)` that would make this dynamic ability possible.

I'd be glad to help work on some solutions to these issues, but I am curious on your leanings to how this should be designed/solved or if you even consider these to be issues. Let me know what I can do.

**Comment by jshcrowthe** [comment]
@liechtym, yeah the modal attribute not being toggleable is a known limitation that I haven't gotten around to addressing yet. I will see if I can get to it tonight.

To your second point about the dynamic content, I just tried a trivial example and it seemed to work fine...

```html
<dialog-el id='dialog'>
  ...
</dialog-el>
<template id='template'>
  <h1>Custom Template</h1>
</template>
```

```javascript
var template = document.querySelector('#template');
var dialog = document.querySelector('#dialog');

// Clear the contents of the dialog
while (dialog.lastChild) { dialog.removeChild(dialog.lastChild) }

dialog.appendChild(document.importNode(template.content, true));
```

Here is a JSBin with it all side by side: http://jsbin.com/wupefa/6/edit?html,output

**Comment by jshcrowthe** [comment]
I have fixed the issue in #24 and also added a getter/setter for the `modal` attribute.

Give the PR a look. Both of the use cases you mentioned before should now be working. You can also do the following:

```html
<button id="button">Open Modal</button>
<dialog-el id="modal"></dialog-el>

<script>
  var modal = document.querySelector('#modal');
  var someCondition = *smart algorithm*;
  // Doesn't work because dialog-el's 'createdCallback' has already initialized the template
  if (someCondition) modal.modal = true;
  document.querySelector('#button').addEventListener('click', function(event){
    modal.show();
  });
</script>
```

**Comment by liechtym** [comment]
The PR looks good to me. And thanks for the clear example on how to update content dynamically!

**Comment by jshcrowthe** [comment]
Closed in #24

**Comment by liechtym** [comment]
Thank you much! This is very helpful!
